includes/api/class-wc-rest-network-orders-controller.php

```diff
@@ -0,0 +1,154 @@
+			'type'        => 'string',
+			'context'     => array( 'view' ),
+			'readonly'    => true,
+		);
+		$schema['properties']['buyer'][] = array(
+			'description' => __( 'Name of the buyer for the order', 'woocommerce' ),
+			'type'        => 'string',
+			'context'     => array( 'view' ),
+			'readonly'    => true,
+		);
+		$schema['properties']['status_name'][] = array(
+			'description' => __( 'Order Status', 'woocommerce' ),
+			'type'        => 'string',
+			'context'     => array( 'view' ),
+			'readonly'    => true,
+		);
+		$schema['properties']['formatted_total'][] = array(
+			'description' => __( 'Order total formatted for locale', 'woocommerce' ),
+			'type'        => 'string',
+			'context'     => array( 'view' ),
+			'readonly'    => true,
+		);
+
+		return $schema;
+	}
+
+	/**
+	 * Does a permissions check for the proper requested blog
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return bool $permission
+	 */
+	public function network_orders_permissions_check( $request ) {
+		$blog_id = $request->get_param( 'blog_id' );
+		$blog_id = ! empty( $blog_id ) ? $blog_id : get_current_blog_id();
+
+		switch_to_blog( $blog_id );
+
+		$permission = $this->get_items_permissions_check( $request );
+
+		restore_current_blog();
+
+		return $permission;
+	}
+
+	/**
+	 * Get a collection of orders from the requested blog id
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_REST_Response
+	 */
+	public function network_orders( $request ) {
+		$blog_id = $request->get_param( 'blog_id' );
+		$blog_id = ! empty( $blog_id ) ? $blog_id : get_current_blog_id();
+
+		switch_to_blog( $blog_id );
+
+		add_filter( 'woocommerce_rest_orders_prepare_object_query', array( $this, 'network_orders_filter_args' ) );
+		$items = $this->get_items( $request );
+		remove_filter( 'woocommerce_rest_orders_prepare_object_query', array( $this, 'network_orders_filter_args' ) );
+
+		foreach ( $items->data as &$current_order ) {
+			$order = wc_get_order( $current_order['id'] );
+
+			$current_order['blog'] = get_blog_details( get_current_blog_id() );
+			$current_order['edit_url'] = get_admin_url( $blog_id, 'post.php?post=' . absint( $order->get_id() ) . '&action=edit' );
+			/* translators: 1: first name 2: last name */
+			$current_order['buyer'] = trim( sprintf( _x( '%1$s %2$s', 'full name', 'woocommerce' ), $order->get_billing_first_name(), $order->get_billing_last_name() ) );
+			$current_order['status_name'] = wc_get_order_status_name( $order->get_status() );
+			$current_order['formatted_total'] = $order->get_formatted_order_total();
+		}
+
+		restore_current_blog();
+
+		return $items;
+	}
+
+	/**
+	 * Filters the post statuses to on hold and processing for the network order query.
+	 *
+	 * @param array $args Query args.
+	 *
+	 * @return array
+	 */
+	public function network_orders_filter_args( $args ) {
+		$args['post_status'] = array(
```

Comment:
If we allowed this to be passed in as an arg to the API, the widget could be improved in the future to allow status filtering.

Reply:
Thought about this, and originally decided to keep it simple, and not open up the endpoint (for now) to more use cases than I needed. I can update this.

Reply:
Actually, I did try and implement this initially, but ran into a case where the current API only supports one status - [see here](https://github.com/woocommerce/woocommerce/blob/master/includes/api/class-wc-rest-orders-controller.php#L351)

In order to change to allowing multiple statuses, the args for the endpoint would need to be changed to `type => array`. I didn't make this change because I didn't want to potentially break or change existing API functionality.

You mentioned there is a v3 of the API in the works - Possibly target array support for `status` on the V3 api instead of just a single string value?

Reply:
Yup good call. I've logged separately. https://github.com/woocommerce/wc-api-dev/issues/73